### ISSUE TITLE
Position issue with galimage or lmpatch or bmp2gal or photoshopped pix when reinserting last bonus banner.

### ISSUE THREAD

**Issue by H0R1Z0N** [pmrowla/pylivemaker#65]
* PyLiveMaker Probably Around may 2 
* Python, latest on stable Arch Linux
* Arch Linux Latest
* AGLS

### Description

I finally got the final picture to reinsert back into the game, which's the "Visual Description/Voice Actress" banner for the bonus free talk. This issue might be related to either galimage, lmpatch or bmp2gal or a mix of those and the method to convert png to bmp.

To resume the issue, when i got the new picture, converted it back and everything, the position of it's wrong when i patch that one. Since that banner's the only one with a weird position, somewhere along the reinsertion, the position code of the picture's handled wrong by something. Instead of having the banner at it normal top right of the sprite, it get push down to the same position than the sprite over it. Now, if i put the og banner in JP instead of the translated one, the position's fine. It did not happen with any other banner either. And this is kinda a cluster fuck to know where the breaking come.

https://media.discordapp.net/attachments/692957773855588382/715429643552817193/unknown.png
https://media.discordapp.net/attachments/692957773855588382/715430007962468392/unknown.png

Reply so i know you will be expecting my email with the full game translation to be able to diagnose it, i have not deleted the backup of the last lmpatch either. I tried multiple method of converting the picture, it might also be the edited picture lacking something or IDK in the picture's code

**Comment by pmrowla** [comment]
Can you just send me the translated png and bitmap you used with galconvert? I shouldn't need the full game, and I have the original .gal image already. I'm guessing that the dimensions of the translated .gal image you ended up with are wrong.

**Comment by H0R1Z0N** [comment]
https://mega.nz/file/WdQX0aoL#34lQjvZL85W9nqN4XcTsZBWBw97V8N3mgS31T0Ah1Ro
I need to precise my build's not up to date but the lmpatch was kinda not updated for that in the history.
The meh's the decent gal file, the bmp's the one resaved inside MS Paint and the bmp copy's one of the multiple conversion i used who did not save correctly (hence why i had to resave using MS Paint) but that what i did for every other picture which worked fine.

**Comment by pmrowla** [comment]
Yeah so the issue is that the original gal file is an 800x600 image (the size of the entire agls screen) with the banner in the correct location, and the rest of the image transparent.

The translated image is just the 67x149 banner itself with the transparent parts cropped out. So the game puts your translated version in the default position (centered and aligned to the bottom of the screen).

what you need is 2 separate bmps that are both 800x600, one with the banner you want (in the correct place) and one black and white only that is used as a mask (so bmp2gale knows what parts of the image are supposed to be transparent) as described in the [LM tutorial](https://pylivemaker.readthedocs.io/en/latest/_static/LiveNovel/tutorial01.html)

basically you need to drag both of the images from the attached zip into BmpToGale together to get the correct result
[img-fixed.zip](https://github.com/pmrowla/pylivemaker/files/4693554/img-fixed.zip)

**Comment by pmrowla** [comment]
this is also why your translated buttons have white pixels in the rounded off corners that probably don't show up in the original, they should really be masked areas as well to make them transparent.

**Comment by H0R1Z0N** [comment]
Do you know how to make the mask easily for the button ?

**Comment by H0R1Z0N** [comment]
Also thank you for the knowledge about the mask and for the file.

**Comment by pmrowla** [comment]
hmm, I just did it manually in photoshop for this.

but if/when I have time to get to #36 the mask won't be needed any more, since PNG and GAL both support transparency properly, meaning we don't need the intermediate BMP format with a separate mask. (the problem with BMP is that it doesn't support transparency, so you need the separate image + mask files to use with bmp2gale)

**Comment by H0R1Z0N** [comment]
Ok, thank you, i will wait then since i can probably do a double release instead or even a triple release of vn.

**Comment by LioMajor** [comment]
@H0R1Z0N in the time being, you can use [Gale](https://graphicsgale.com/files/GalePortable20821.zip) for 2.x.

You can set it to english:
![image](https://user-images.githubusercontent.com/591183/83111032-44693080-a0c4-11ea-9ca1-35afa017bc2f.png)

Start GaleBrowse with japanese (emulated or not) to support japanese file names.

1. Browse to the picture you want to edit
2. Press Enter to view in full size
3. Press Enter to view in GraphicsGale
4. File > Save as... > Switch output to png
5. Add the Alpha Channel
6. Save the PNG
7. Edit with your desired tool (photoshop)
8. Convert it back to .gal using GraphicsGale

You can assign shortcuts and for multiple files use batch convert.

If you have a game with v1.x gal files, you need to use the old GraphicsGale or the game stucks loading the picture.

**Comment by H0R1Z0N** [comment]
Thank you, was able to fix the other picture thank to your software post. Thank you very much. The project's now 100% good for release... But i got a tradition in my translation to put a special thank after the credit by inserting new code. So i am currently doing that. I wanted to release before the 1st of June, but i might be a bit late but not for too long. I will certainly give credit where people deserve it, aka for anything, lol.

**Comment by pmrowla** [comment]
I added an `lmbmp` utility that will generate image + mask BMPs to be used with BmpToGale.

one other thing to note is that you have to select "normal compression" (`通常圧縮`) for BmpToGale to handle masks properly (it defaults to uncompressed which doesn't use the mask for some reason):
![bmp2gale](https://user-images.githubusercontent.com/651988/83323163-ee4ce680-a297-11ea-9753-09648db24dd0.PNG)

**Comment by pmrowla** [comment]
closing this since there's workarounds available until we have an actual gal encoder
